src/Magento/FunctionalTestingFramework/DataGenerator/Handlers/SecretStorage/VaultStorage.php

```diff
@@ -120,4 +134,82 @@ private function authenticated()
         }
         return false;
     }
+
+    /**
+     * Read vault token from file system
+     *
+     * @return void
+     * @throws TestFrameworkException
+     */
+    private function readVaultTokenFromFileSystem()
+    {
+        // Find user home directory
+        $homeDir = getenv('HOME');
+        if ($homeDir === false) {
+            // If HOME is not set, don't fail right away
+            $homeDir = '~/';
+        } else {
+            $homeDir = rtrim($homeDir, '/') . '/';
```

Comment:
It's unnecessary to do rtrim and then add slash afterwards. 
I do suggest to use `realpath` php function which will return you proper path to directory you trying to reach.

Reply:
That's a good idea. I will use realpath().